src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -925,7 +925,14 @@ internal SecurityStatusPal Decrypt(byte[]? payload, ref int offset, ref int coun
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            return SslStreamPal.DecryptMessage(_securityContext!, payload!, ref offset, ref count);
+            SecurityStatusPal status = SslStreamPal.DecryptMessage(_securityContext!, payload!, ref offset, ref count);
+            if (NetEventSource.Log.IsEnabled() && status.ErrorCode == SecurityStatusPalErrorCode.OK)
+            {
+                ReadOnlyMemory<byte> buffer = new ReadOnlyMemory<byte>(payload, offset, count);
```

Comment:
You don't have to convert the buffer into `Memory` just for the logging. `DumpBuffer` has `byte[], offset, count` [overload](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Net/Http/aspnetcore/NetEventSource.Common.cs#L307-L307).
And the same thing for `"Decrypt"`.

Reply:
updated.